test/forms/sections/firstTransporter/FirstTransporterVatFormProviderSpec.scala

```diff
@@ -16,38 +16,95 @@
 
 package forms.sections.firstTransporter
 
+import config.Constants.NONGBVAT
+import forms.ONLY_ALPHANUMERIC_REGEX
 import forms.behaviours.StringFieldBehaviours
+import models.sections.firstTransporter.FirstTransporterVatModel
 import play.api.data.FormError
 
 class FirstTransporterVatFormProviderSpec extends StringFieldBehaviours {
 
-  val requiredKey = "firstTransporterVat.error.required"
+  val requiredKey = "firstTransporterVat.error.input.required"
   val lengthKey = "firstTransporterVat.error.length"
-  val maxLength = 12
+  val maxLength = 14
 
   val form = new FirstTransporterVatFormProvider()()
 
-  ".value" - {
+  "when binding 'Yes'" - {
 
-    val fieldName = "value"
+    "when vat contains invalid characters" - {
 
-    behave like fieldThatBindsValidData(
-      form,
-      fieldName,
-      "0" * maxLength
-    )
+      "must error when binding the form" in {
 
-    behave like fieldWithMaxLength(
-      form,
-      fieldName,
-      maxLength = maxLength,
-      lengthError = FormError(fieldName, lengthKey, Seq(maxLength))
-    )
+        val boundForm = form.bind(Map(
+          FirstTransporterVatFormProvider.hasVatField -> "true",
+          FirstTransporterVatFormProvider.vatNumberField -> "<"
+        ))
 
-    behave like mandatoryField(
-      form,
-      fieldName,
-      requiredError = FormError(fieldName, requiredKey)
-    )
+        boundForm.errors mustBe Seq(FormError(
+          FirstTransporterVatFormProvider.vatNumberField,
+          FirstTransporterVatFormProvider.vatNumberInvalid,
+          Seq(ONLY_ALPHANUMERIC_REGEX)
+        ))
+      }
+    }
+
+    "when vat number is too long" - {
+
+      "must error when binding the form" in {
+
+        val boundForm = form.bind(Map(
+          FirstTransporterVatFormProvider.hasVatField -> "true",
+          FirstTransporterVatFormProvider.vatNumberField -> "a" * (FirstTransporterVatFormProvider.vatNumberMaxLength + 1)
+        ))
+
+        boundForm.errors mustBe Seq(FormError(
+          FirstTransporterVatFormProvider.vatNumberField,
+          FirstTransporterVatFormProvider.vatNumberLength,
+          Seq(FirstTransporterVatFormProvider.vatNumberMaxLength)
+        ))
+      }
+    }
+
+    "when vat number is valid" - {
+
+      "must bind the form successfully when true with value" in {
+
+        val boundForm = form.bind(Map(
+          FirstTransporterVatFormProvider.hasVatField -> "true",
+          FirstTransporterVatFormProvider.vatNumberField -> "123456789"
+        ))
+
+        boundForm.errors mustBe Seq()
```

Comment:
This check isn't needed. Because, on line 80, the form is being checked for a successful bind, implicitly there will be no errors. Very minor though 😄